src/server/scripts/Kalimdor/zone_durotar.cpp

```diff
@@ -177,9 +178,571 @@ struct npc_mithaka : ScriptedAI
     bool _shipInPort;
 };
 
+// Echo Isles
+
+class spell_summon_zuni : public SpellScript
+{
+    void SetDest(SpellDestination& dest)
+    {
+        dest.Relocate({ - 1173.4531f, -5266.401f, 0.85905945f, 0.0f });
+    }
+
+    void Register() override
+    {
+        OnDestinationTargetSelect += SpellDestinationTargetSelectFn(spell_summon_zuni::SetDest, EFFECT_0, TARGET_DEST_NEARBY_ENTRY);
+    }
+};
+
+enum TikiTarget
+{
+    SPELL_ARCANE_MISSILES_TRAINER = 83470,
+    SPELL_TIKI_TARGET_VISUAL_1    = 71064,
+    SPELL_TIKI_TARGET_VISUAL_2    = 71065,
+    SPELL_TIKI_TARGET_VISUAL_3    = 71066,
+    SPELL_TIKI_TARGET_DEATH       = 71240
+};
+
+uint32 const TiKiTargetMask[3] = { SPELL_TIKI_TARGET_VISUAL_1, SPELL_TIKI_TARGET_VISUAL_2, SPELL_TIKI_TARGET_VISUAL_3 };
+
+struct npc_tiki_target : public ScriptedAI
```

Comment:
any struct belonging to a given zone_name must have such name in its name

Reply:
Done